packages/theme/src/theme/_forms.scss

```diff
@@ -307,30 +327,47 @@ form {
 	&-control {
 		+ label {
 			position: absolute;
-			top: 1.5rem;
+			top: 2.2rem;
 			left: 0;
-			font-size: 1.5rem;
+			font-size: $font-size-base;
 			transition: opacity 0.1s linear, font-size 0.1s linear, top 0.1s linear;
 			pointer-events: none;
 		}
+
+		&.input-sm + label{
+			font-size: $font-size-small;
+		}
+		&.input-lg + label{
+			top: 2.6rem;
+			font-size: $font-size-large;
+		}
+
 	}
 
 	&-control:focus,
-	&-control:disabled,
-	&-control[value]:not([value='']),
-	&-control:not(:empty),
+	&-control[placeholder]:not([placeholder='']),
+		&-control:not(:empty),
 	input[type='date'],
 	input[type='datetime-local'] {
 		+ label {
 			top: 0;
-			font-size: 1.125rem;
-			opacity: 1 !important;
+			font-size: $font-size-small;
 		}
 	}
 
-	&-control[placeholder]:not([placeholder='']) {
+	&-control:focus,
+	{
 		+ label {
-			opacity: 0;
+			opacity: 1 !important;
```

Comment:
done. I'm going to see that for the element textarea, select we can't detect if the user has set a value. I'm going to see with @maximenicoul if the state is relevant.